File: 1001-n-repeated-element-in-size-2n-array/n-repeated-element-in-size-2n-array.java
class Solution {
    public int repeatedNTimes(int[] nums) 
    {
        HashMap<Integer,Integer> map = new HashMap<>();
        int ans = 0;
        int n = nums.length / 2;
        for(int i:nums)
        {
            map.put(i,map.getOrDefault(i,0) + 1 );
        }

        for( Map.Entry<Integer,Integer> entry : map.entrySet())
        {
            if (entry.getValue() == n )
            {
                ans = entry.getKey();
                break;
            }
        }
        return ans;
        
    }

}

File: 1001-n-repeated-element-in-size-2n-array/n-repeated-element-in-size-2n-array.cpp
class Solution {
public:
    int repeatedNTimes(vector<int>& nums) {
        int n = nums.size() / 2;
        unordered_map<int,int> um;
        int ans = 0;

        for(int i:nums)
        {
            um[i]++;
        }

        for(auto i : um)
        {
            if (i.second == n) 
            {
                ans = i.first;
                return ans;
            }
        }
        return ans;
        
    }
};